### ISSUE TITLE
Add ability to install multiple package versions

### ISSUE THREAD

**Issue by wizarden1** [PowerShellOrg/cChoco#140]
I rewrite cChocoPackageInstall and now it can be used for install multiple version of app. It use '-m' switch for it
Pros:
- can install multiple version
- check all versions for complaint
- can uninstall unused and install required versions
Cons:
- Upgrade process doesn't work, only install/uninstall

If this interesting, i will put code here

**Comment by pauby** [comment]
If the upgrade process does not work then that breaks functionality of the resource.

**Comment by wizarden1** [comment]
[cChocoPackageInstall.zip](https://github.com/chocolatey/cChoco/files/4291213/cChocoPackageInstall.zip)
No, I mean if you set multiple versions, module use functions UninstallPackage, InstallPackage and not Upgrade-Package

**Comment by wizarden1** [comment]
full module, ignore zip extension it's nuget package
[cchoco.2.4.1.nupkg.zip](https://github.com/chocolatey/cChoco/files/4291220/cchoco.2.4.1.nupkg.zip)
And 2.4.1 it's not based on you 2.4.1. When I write it, 2.4.1 hasn't been exist, it based on 2.4.0 from powershell gallery

**Comment by pauby** [comment]
Submit a PR and I can look at the changes.

**Comment by pauby** [comment]
PR #143 submitted.

**Comment by pauby** [comment]
@wizarden1 I was never keen on multiple installs (experience) but this comment in the [Choco repository](https://github.com/chocolatey/choco/issues/2046#issuecomment-629578136) has me thinking about this again. What I don't want to happen is for us to cause more issues than it solves.

Maybe a HUGE caveat around it? Thoughts?

**Comment by wizarden1** [comment]
You are right, this function is two side sword. From one perspective it'll allow to install multiple version of software that support this function. From other side this function will do strange thing with software that not support multiple versions on one OS. And the main question is, how choco per se work with "-m" parameter. As I see this function give some pros and con.
Pros
- ability to install multiple version if necessary

Con
- can be used with software that doesn't support this possibility. And get permanent inconstant of state.

May be put some warning about application that can't be used with multiple versions. And use it for your own risk.

But if the person is an idiot none of restriction measurements can help.

Basically multi version just use choco with "-m" parameter and it depends of what choco will do.

What of HUGE caveat around it you see?
And I seen many request about this function from community.

**Comment by pauby** [comment]
PR #143 was closed due to being stale.
